dotnet/CoreLib/Handlers/SummarizationHandler.cs

```diff
@@ -95,27 +97,33 @@ public class SummarizationHandler : IPipelineStepHandler
                             var destFile = uploadedFile.GetHandlerOutputFileName(this);
                             await this._orchestrator.WriteFileAsync(pipeline, destFile, new BinaryData(summary), cancellationToken).ConfigureAwait(false);
 
-                            summaryFiles.Add(destFile, new DataPipeline.GeneratedFileDetails
+                            lock (summaryFiles)
```

Comment:
See my comment [here](https://github.com/microsoft/kernel-memory/pull/147/files#r1388085382).

Reply:
fix, but I did this lock, becase this dictionary is private end exisit only in this method, but you are right. it's safe to have separte variable for this